src/pages/ImportData.js

```diff
@@ -140,17 +140,18 @@ const ImportData = () => {
         notifyDataImportProgress(jobId)
       }],
       ['snapshot', async () => {
+        // Normalize the title:
+        // Importing snapshot will throw an "enum" error if the title has any spaces or special characters
+        // Replace all whitespace characters with _
+        // Then replace all non alphanumeric characters with nothing
+        const normalizedTitle = _.flow(
+          _.replace(/\s/g, '_'),
+          _.replace(/[^A-Za-z0-9-_]/g, '')
+        )(snapshotName)
+
         if (!_.isEmpty(snapshots)) {
           const responses = await Promise.allSettled(
             _.map(({ title, id, description }) => {
-              // Normalize the title:
-              // Importing snapshot will throw an "enum" error if the title has any spaces or special characters
-              // Replace all whitespace characters with _
-              // Then replace all non alphanumeric characters with nothing
-              const normalizedTitle = _.flow(
-                _.replace(/\s/g, '_'),
-                _.replace(/[^A-Za-z0-9-_]/g, '')
-              )(title)
```

Comment:
This refactor would break importing from Data Catalog.

Note that here, in the deleted code, it is normalizing the variable `title` … but in the new code at line 150, it is normalizing the variable `snapshotName`. Thus - here inside the if clause - it'll end up with the wrong value.

My suggestion is to extract a new _function_ for normalization. This could go at the root of the `ImportData` component, parallel to `const filteredTemplates`, `const onImport`, etc:
```javascript
  // Normalize a snapshot name:
  // Importing snapshot will throw an "enum" error if the name has any spaces or special characters
  // Replace all whitespace characters with _
  // Then replace all non alphanumeric characters with nothing
  const normalizeSnapshotName = input => _.flow(
    _.replace(/\s/g, '_'),
    _.replace(/[^A-Za-z0-9-_]/g, '')
  )(input)
```

In this suggestion, note that I've renamed a few variables … but most importantly, it adds the `input => ` syntax, which turns this into a function instead of procedural code. Then, in the top half (Data Catalog) of the if clause you can use: 
```javascript
Ajax().Workspaces.workspace(namespace, name).importSnapshot(id, normalizeSnapshotName(title), description)
```
and in the bottom half (TDR import-by-copy) you can use:
```javascript
Ajax().Workspaces.workspace(namespace, name).importSnapshot(snapshotId, normalizeSnapshotName(snapshotName))
```